### PR TITLE
Fix: rating bugs & text modifs

### PR DESCRIPTION
**Pull request by Meriem-BM**
Related #53 #54 #58 #55 #59 #39

### CODE REVIEW

File: app/comparison/[category]/page.tsx [hunk 1]
```diff
@@ -57,8 +57,8 @@ export default function Home() {
   const { checkLoginFlow } = useAuth();
   const { address, chainId } = useAccount();
 
-  const [rating1, setRating1] = useState<number>(3);
```
Comment (mmahdigh) [inline app/comparison/[category]/page.tsx:60]: However, the default value for stars must be `null` and if users don't rate a project, it should be sent as `null` to our backend server.

File: app/comparison/[category]/page.tsx [hunk 2]
```diff
@@ -326,8 +327,8 @@ export default function Home() {
       data: {
         project1Id: project1!.id,
         project2Id: project2!.id,
-        project1Stars: rating1,
-        project2Stars: rating2,
+        project1Stars: rating1 || 3,
```
Comment (mmahdigh) [inline app/comparison/[category]/page.tsx:330]: Here, it shouldn't be sent as a `3` to the backend if it's a `null`

Comment (mmahdigh) [reply app/comparison/[category]/page.tsx:330]: Keep the rating as `null`at every stage and just show it as a 0 in Line 546. (something like: `rating1 === null ? 0 : rating1`)

Comment (Meriem-BM) [reply app/comparison/[category]/page.tsx:330]: @MoeNick mentioned that it still should be sent as 3 as default to backend if it's null.

**Comment by Meriem-BM** [comment]
I made those modifications @mmahdigh, you can re-review it

**Review by mmahdigh — APPROVED** [review]

**Pull request merged**